Gui: Fix Document Recovery screen

kadet1090 opened the pull request:
This fixes looks of document recovery screen getting rid of wasted space. It also should properly react to resize.

<!--
The FreeCAD community thanks you for your contribution!
By creating a Pull Request you agree to the contributing policy. The complete policy can be found in the root of the source tree (CONTRIBUTING.md) or at https://github.com/FreeCAD/FreeCAD/blob/main/CONTRIBUTING.md

This template provides guidance on creating a PR that can be reviewed and approved as quickly as possible. Comments may be safely deleted.

Unless you know exactly what you're doing, please leave the checkbox 'Allow edits by maintainers' enabled.  This will allow maintainers to help you.
-->

## Issues
Fixes #21194
<!-- link to individual issues this PR closes by referencing the issue number (e.g., fixes #1234, closes #4321). -->

## Before and After Images
| Before | After |
|--------|--------|
| <img width="570" height="590" alt="image" src="https://github.com/user-attachments/assets/7d568637-b2a8-4fae-8bac-00c78d12aa31" /> | <img width="560" height="434" alt="image" src="https://github.com/user-attachments/assets/2a96a950-ea97-4306-9e44-9549b86b79e3" /> | 
<!-- If your proposed changes affect the FreeCAD GUI, add before and after screenshots -->



<!--  Notes on the PR Review Process

The following section describes what the maintainers consider when reviewing your Pull Request.  These items may not require you to take any action.  This information is provided for context. Understanding what we consider will help you prepare your request for speedy approval.

You can find additional documentation about these guidelines in the [Developers handbook](https://freecad.github.io/DevelopersHandbook).

Alignment (Does the PR align with the goals and interests of the project?)
  - Does the PR have at least one issue linked, which this PR closes?
  - Has the conversation on the PR and related issue(s) reached consensus?
  - If the PR affects the GUI, is the Design Working Group (DWG) aware and have they had time to review and comment?
  - If the PR affects the GUI, did the contributor include before/after images?
  - If the PR affects standards and workflow, is the CAD Working Group (CWG) aware and have they had time to review/comment?

Impact (Does the change affect other parts of the project?)
  - Has the impact on documentation been considered and appropriate action taken?
  - Has the impact on translation been considered appropriate action taken?
  - Will the PR affect existing user documents?

Code Quality (Is code well-written and maintainable?)
  - Does the PR warrant a review by the Code Quality Working Group (CQWG)?
  - Does the change include tests?
  - Is the PR rebased on the current main branch with unnecessary commits squashed?

Release (Are there considerations related to release timing?)
  - Has the PR been considered for backporting to the latest release branch?
  - Have the release notes been considered/updated?
  -->

xtemp09 commented:
Might I ask, what the symbol at the bottom of `src/Gui/Stylesheets/FreeCAD.qss` mean?










----









The code contains lines:







```cpp
auto w = qobject_cast<QMdiSubWindow*>(myViewer->parentWidget());
w->close();
```









It does not check the validity of `w`. Does it work if the window is undocked?

kadet1090 commented:
> Might I ask, what the symbol at the bottom of `src/Gui/Stylesheets/FreeCAD.qss` mean?

That there is no end of line on the last line - I added one. Thanks.

> The code contains lines:
> 
> ```c++
> auto w = qobject_cast<QMdiSubWindow*>(myViewer->parentWidget());
> w->close();
> ```
> 
> It does not check the validity of `w`. Does it work if the window is undocked?

Thanks, that was unrelated change commited by mistake - I removed it from the commit.
